src/components/Session/withAuthentication.test.js

```diff
@@ -26,13 +27,29 @@ describe('withAuthentication', () => {
     const firebase = new Firebase();
     render(
       <FirebaseContext.Provider value={firebase}>
-        <DummyComponent />
+        <Router>
+          <DummyComponent />
+        </Router>
       </FirebaseContext.Provider>
     );
     expect(screen.getByText('Test user null')).toBeInTheDocument();
     firebase.login(validEmail, validPassword);
     await flushPromises();
     expect(screen.getByText('Test user set')).toBeInTheDocument();
   });
+  it('Redirect no authenticated user to Landing Page', async () => {
```

Comment:
This does not check if Route is changed. You can test it in App.test.js

Here you can check if the log out resets user.